guides/micronaut-openapi-generator-server/micronaut-openapi-generator-server.adoc

```diff
@@ -0,0 +1,497 @@
+
+common:graal-with-plugins.adoc[]
+
+== Next steps
+
+=== Add Security
+
+We could have defined our security requirements by adding a security scheme to the `library-definition.yaml` file.
+For example, we will add basic authentication:
+
+[source,yaml]
+----
+paths:
+  /search:
+    # ... #
+  /add:
+    post:
+      # ... #
+      security:
+        - MyBasicAuth: [] # <2>
+components:
+  schemas:
+    # ... #
+  securitySchemes:
+    MyBasicAuth: # <1>
+      type: http
+      scheme: basic
+----
+<1> Define a security scheme inside the `components/securitySchemes`. We want to use basic auth for authentication.
+<2> Add the scheme to the paths that you want to secure. In this case we want to restrict access to
+adding books into our library.
+
+NOTE: You can read more about describing various authentication in the
+link:https://swagger.io/docs/specification/authentication/["Authentication and Authorization" OpenAPI guide].
+
+The generator will then annotate such endpoints with Micronaut`s
+link:https://micronaut-projects.github.io/micronaut-security/latest/guide/#secured[Secured] annotation accordingly:
+
+[source,java]
+----
+@Secured(SecurityRule.IS_AUTHENTICATED)
+public Mono<Object> addBook( /* ... */ ){ /* ... */ }
+----
+
+You will then need to implement an
+link:https://micronaut-projects.github.io/micronaut-security/latest/guide/#authenticationProviders[AuthenticationProvider]
+that satisfies your needs. If you want to finish implementing the basic authentication, continue to the
+link:https://guides.micronaut.io/latest/micronaut-security-basicauth.html[Micronaut Basic Auth guide] and replicate
+steps to create the `AuthenticationProvider` and appropriate tests.
+
+NOTE: You can also read link:https://micronaut-projects.github.io/micronaut-security/latest/guide/[Micronaut Security documentation]
+or link:https://micronaut.io/guides[Micronaut guides] about security to learn more about all
+the supported Authorization strategies.
+
+=== Generate Micronaut Client
+
+You can generate Micronaut Client based on the same `library-definition.yaml` file.
+
+Run the following in terminal to create client in the `library-client` folder:
+
+[source,bash]
+----
+java -jar openapi-generator-cli-5.4.0.jar generate \
+    -g java-micronaut-client \
+    -i library-definition.yaml \
+    -o library-client \
+    -p apiPackage=example.micronaut.library.api \
+    -p modelPackage=example.micronaut.library.model \
+    -p build=@build@ \
+    -p test=junit
+----
+
+=== Learn Micronaut
+
+To learn more about Micronaut framework and its features visit
+link:https://micronaut.io/docs/[Micronaut documentation].
+
+=== Generate User-Friendly Documentation
+
+You can generate documentation in html file inside the `html-docs/` folder by running
+[source,bash]
+----
+java -jar openapi-generator-cli-5.4.0.jar generate \
+    -g html2 \
+    -i library-definition.yaml \
+    -o html-docs
+----
+
+common:helpWithMicronaut.adoc[]
```

Comment:
Delete helpWithMicronaut.adoc, that's only for guides written by Object Computing